yt_dlp/extractor/twitch.py

```diff
@@ -1135,81 +1161,68 @@ def _real_extract(self, url):
             'sig': clip['playbackAccessToken']['signature'],
             'token': clip['playbackAccessToken']['value'],
         }
-
-        data = self._download_base_gql(
-            video_id, {
-                'query': '''{
-  clip(slug: "%s") {
-    broadcaster {
-      displayName
-    }
-    createdAt
-    curator {
-      displayName
-      id
-    }
-    durationSeconds
-    id
-    tiny: thumbnailURL(width: 86, height: 45)
-    small: thumbnailURL(width: 260, height: 147)
-    medium: thumbnailURL(width: 480, height: 272)
-    title
-    videoQualities {
-      frameRate
-      quality
-      sourceURL
-    }
-    viewCount
-  }
-}''' % video_id}, 'Downloading clip GraphQL', fatal=False)  # noqa: UP031
-
-        if data:
-            clip = try_get(data, lambda x: x['data']['clip'], dict) or clip
+        asset_default = traverse_obj(clip, ('assets', 0), {})
+        asset_portrait = traverse_obj(clip, ('assets', 1), {})
+        game = traverse_obj(clip, ('game', 'displayName'))
 
         formats = []
-        for option in clip.get('videoQualities', []):
-            if not isinstance(option, dict):
-                continue
-            source = url_or_none(option.get('sourceURL'))
-            if not source:
-                continue
-            formats.append({
-                'url': update_url_query(source, access_query),
-                'format_id': option.get('quality'),
-                'height': int_or_none(option.get('quality')),
-                'fps': int_or_none(option.get('frameRate')),
-            })
+        for quality in traverse_obj(asset_default, ('videoQualities'), default=[]):
+            if source := traverse_obj(quality, ('sourceURL', {url_or_none})):
+                formats.append({
+                    'url': update_url_query(source, access_query),
+                    'format_id': quality.get('quality'),
+                    'height': int_or_none(quality.get('quality')),
+                    'fps': float_or_none(quality.get('frameRate')),
+                    'aspect_ratio': float_or_none(asset_default.get('aspectRatio')),
+                })
```

Comment:
`from yt_dlp.utils.traversal import value, traverse_obj`
```suggestion
        default_aspect_ratio = float_or_none(asset_default.get('aspectRatio'))
        formats.extend(traverse_obj(asset_default, ('videoQualities', lambda _, v: url_or_none(v['sourceURL']), {
            'url': ('sourceURL', {update_url_query(query=access_query)}),
            'format_id': ('quality, {str}),
            'height': ('quality', {int_or_none}),
            'fps': ('frameRate', {float_or_none}),
            'aspect_ratio': {value(default_aspect_ratio)},
        })))
```